### DISCUSSION TITLE
Geo IP

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#397] [Show and tell]
With 0.50 we get the possibility to see the country of origin of connections (displayed under Admin > Monitoring), but also to allow or block some countries.

<img width="417" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/a571458d-3468-4fb3-a215-2720f11673fe">

**Comment by W-i-n-7** [comment]
awesome

**Comment by mcodr** [comment]
It seems like when GEO IP feature is enabled, and the "When country cannot be determined is set to: **Block**", you are no longer able to access the site/admin panel locally using direct local IP.

Any way, to still allow it?

**Comment by rejetto** [comment]
of course with local you don't mean localhost, but local network.
that's expected, a "local network" is not always trusted, it can be your college campus, filled with people you don't know.

but maybe it's better if the geo-firewall doesn't act on the LAN, leaving that to the general "block" feature. That sounds like more flexible.

**Comment by W-i-n-7** [comment]
first thing that came to mind was that there could be a "only allow the following local ip:s" setting
but i dont know how good that would be

**Comment by rejetto** [comment]
another option is not really needed, if can block local ips using "blocked ip" feature

**Comment by W-i-n-7** [comment]
except you wouldnt know an attacker's ip and local ips change due to dhcp

**Comment by mcodr** [comment]
in my case, I got an old netbook computer dedicated to run the HFS server at home.
I would prefer to use the local ip to check admin panel on my main computer.

Perhaps an "allowed ip" feature that can overwrite Geo IP?

**Comment by W-i-n-7** [comment]
or a toggle to affect or not local ips

**Comment by rejetto** [comment]
> except you wouldnt know an attacker's ip and local ips change due to dhcp

geo is not blockin one specific ip but the whole lan, so i don't understand your point.
if you want to block the lan you block 192.168.0.* for example, and you are done, you don't need to know other's ip.

my point is that you at the moment CANNOT exclude the lan.
with the other way i described you CAN include lan if you want.
geo has very little to do with a lan.

**Comment by rejetto** [comment]
would you try it and give me a feedback?
just right click on "check for updates" and paste this link https://github.com/rejetto/hfs/files/14157522/hfs-windows.zip

**Reply by mcodr** [reply]
running 0.52.0 alpha3 now, using direct IP address didn't work on both machine. 
Only http://localhost/( on server) and duckdns domain address works.

**Reply by rejetto** [reply]
what kind of "direct ip address" ?

**Reply by W-i-n-7** [reply]
pretty sure he means accessing the server with local ip

**Reply by W-i-n-7** [reply]
i think you should allow local ip access when the country cant be determined

**Reply by mcodr** [reply]
yes, that is exactly what i meant. thank you. 
for example - http://192.168.1.10:8080/~/admin/#/ does not work on 192.168.1.11 even when on 192.168.1.10

**Reply by W-i-n-7** [reply]
i access over local ip alot too

**Reply by rejetto** [reply]
i have this configuration, and i'm correctly connecting on to 192.168.0.3, so I don't know what's going on there
<img width="179" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/a90b8140-a5f4-4193-b311-8b51614525b8">

@W-i-n-7  do you have the same problem as he?

**Reply by mcodr** [reply]
I got it to work now! Thank you!
I had to uncheck this settings which is funny.
<img width="662" alt="image" src="https://github.com/rejetto/hfs/assets/7112407/40bd64ae-3b4f-4d4d-ba33-0df833e58258">

**Reply by W-i-n-7** [reply]
oh that makes sense
